fastlane_core/lib/fastlane_core/device_manager.rb

```diff
@@ -40,20 +41,59 @@ def all
         return @devices
       end
 
-      def clear_cache
-        @devices = nil
-      end
+      def connected_devices(requested_os_type = "")
+        Helper.log.info "Fetching available connected devices" if $verbose
 
-      # Reset all simulators of this type
-      def reset_all
-        all.each(&:reset)
+        device_types = []
+        if requested_os_type == "tvOS"
+          device_types = ["AppleTV"]
+        elsif requested_os_type == "iOS"
+          device_types = ["iPhone", "iPad"]
+        end
```

Comment:
Suggestion: This can be tightened up a bit like:

```ruby
device_types = if requested_os_type == "tvOS"
                 ["AppleTV"]
               elsif requested_os_type == "iOS"
                 ["iPhone", "iPad"]
               else 
                 []
               end
```